### DISCUSSION TITLE
When using the BSELoss function, we pass the ground truth as y_train which contains either class 0 or 1 so the predictions should also be predicted classes and not class probabilities right?

### DISCUSSION THREAD

**Discussion by Blurr99** [mrdbourke/pytorch-deep-learning#1157] [General]
So, in this video https://www.youtube.com/watch?v=LyJtbe__2i0, at 10:47:28, Daniel uses:
```
loss = loss_fn(torch.sigmoid(y_logits),  y_train)
```

but after the logits pass through the sigmoid function they turn into class probabilities right? So how can they be compared to y_train which contains either 0 or 1. Shouldn't it be:
```
loss = loss_fn(torch.round(torch.sigmoid(y_logits),  y_train))
```

**Comment by pem131** [comment]
Greetings Blurr99,

The code `loss = loss_fn(torch.sigmoid(y_logits),  y_train)`  as Daniel used is correct.

You can see in the [documentation](https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html) (math included) that BCELoss measures the Binary Cross Entropy between the target and the _input probabilities_. Therefore, you shall feed the probabilities _as they are_ (i.e., no rounding) into the loss function.

I hope this answers your question.

Best wishes.
